app/src/main/kotlin/com/wire/android/ui/home/conversations/folder/NewConversationFolderScreen.kt

```diff
@@ -0,0 +1,203 @@
+import com.wire.android.ui.common.rememberBottomBarElevationState
+import com.wire.android.ui.common.rememberTopBarElevationState
+import com.wire.android.ui.common.scaffold.WireScaffold
+import com.wire.android.ui.common.textfield.DefaultText
+import com.wire.android.ui.common.textfield.WireTextField
+import com.wire.android.ui.common.textfield.WireTextFieldState
+import com.wire.android.ui.common.textfield.maxLengthWithCallback
+import com.wire.android.ui.common.topappbar.NavigationIconType
+import com.wire.android.ui.common.topappbar.WireCenterAlignedTopAppBar
+import com.wire.android.ui.home.settings.account.displayname.ChangeDisplayNameViewModel.Companion.NAME_MAX_COUNT
+import com.wire.android.ui.theme.WireTheme
+import com.wire.android.ui.theme.wireColorScheme
+import com.wire.android.ui.theme.wireDimensions
+import com.wire.android.util.ui.PreviewMultipleThemes
+import com.wire.android.util.ui.SnackBarMessageHandler
+
+@RootNavGraph
+@WireDestination
+@Composable
+fun NewConversationFolderScreen(
+    navigator: Navigator,
+    resultNavigator: ResultBackNavigator<String>,
+    viewModel: NewFolderViewModel = hiltViewModel()
+) {
+
+    LaunchedEffect(viewModel.folderNameState.folderId) {
+        if (viewModel.folderNameState.folderId != null) {
+            resultNavigator.navigateBack(viewModel.folderNameState.folderId!!)
+        }
+    }
+
+    Content(
+        textState = viewModel.textState,
+        state = viewModel.folderNameState,
+        onContinuePressed = {
+            viewModel.createFolder(viewModel.textState.text.toString())
+        },
+        onBackPressed = navigator::navigateBack
+    )
+
+    SnackBarMessageHandler(viewModel.infoMessage)
+}
+
+@Composable
+private fun Content(
+    textState: TextFieldState,
+    state: FolderNameState,
+    onContinuePressed: () -> Unit,
+    onBackPressed: () -> Unit,
+    modifier: Modifier = Modifier
+) {
+    val scrollState = rememberScrollState()
+    with(state) {
+        WireScaffold(
+            modifier = modifier,
+            topBar = {
+                WireCenterAlignedTopAppBar(
+                    elevation = scrollState.rememberTopBarElevationState().value,
+                    onNavigationPressed = onBackPressed,
+                    navigationIconType = NavigationIconType.Back(),
+                    title = stringResource(id = R.string.label_new_folder)
+                )
+            }
+        ) { internalPadding ->
+            Column(
+                modifier = Modifier
+                    .padding(internalPadding)
+                    .fillMaxSize()
+            ) {
+                Column(
+                    modifier = Modifier
+                        .weight(weight = 1f, fill = true)
+                        .fillMaxWidth()
+                        .verticalScroll(scrollState)
+                ) {
+                    val keyboardController = LocalSoftwareKeyboardController.current
+
+                    Spacer(modifier = Modifier.weight(0.5f))
```

Comment:
question: on designs this input is supposed to be centered, but here spacer above it has 0.5 weight and below 1.0 - shouldn't both be equal? Or maybe even just add `verticalArrangement = Arrangement.Center,` to the column without spacers 🤔

Reply:
Yup removed not necessary column